server/src/main.rs

```diff
@@ -1,17 +1,110 @@
+use futures_lite::stream::StreamExt;
+
 use crate::model::QueryRoot;
 use crate::merkle::get_new_merkle_tree;
 use crate::routes::{graphql_handler, graphql_playground, health};
 use async_graphql::{EmptySubscription, Schema};
 use axum::{extract::Extension, routing::get, Router, Server};
+use lapin::publisher_confirm::Confirmation;
 use std::sync::Arc;
 use futures::lock::Mutex;
 use model::MutationRoot;
+use tokio_postgres::NoTls;
+use lapin::{
+    options::*, types::FieldTable, BasicProperties, Connection, ConnectionProperties,
+};
 mod model;
 mod routes;
 mod merkle;
 
 #[tokio::main]
 async fn main() {
+    /////////////// experimental postgresql integration ///////////////
+    // Connect to the database.
+    let (client, connection) = tokio_postgres::connect("host=localhost user=dev dbname=arcpay", NoTls).await.unwrap();
+
+    // The connection object performs the actual communication with the database,
+    // so spawn it off to run on its own.
+    tokio::spawn(async move {
+        if let Err(e) = connection.await {
+            eprintln!("connection error: {}", e);
+        }
+    });
+
+    let rows = client
+        .query("SELECT * FROM state_tree", &[])
+        .await.unwrap();
```

Comment:
I feel like this could fail, since we're not necessarily waiting for a connection to the database to complete - we're awaiting the connection to complete in a separate thread.

Reply:
your reasoning looks correct, but there has to be some internals we may be missing since I copied this from the official reference code: https://docs.rs/tokio-postgres/latest/tokio_postgres/